pytorch_lightning/loops/dataloader/evaluation_loop.py

```diff
@@ -80,6 +81,10 @@ def reset(self) -> None:
             self._max_batches = [self._max_batches] * len(self.dataloaders)
 
         super().reset()
+        # when restarting, if we are running `validate` or `test` twice, since there's no concept of `max_epochs` we
+        # need to reset the current state when the loop has finished running
+        if self.done and self.trainer.state.fn != TrainerFn.FITTING:
+            self.dataloader_progress.reset_on_run()
```

Comment:
I considered putting this in the super implementation, but I don't think the `state.fn` check makes sense there.

Reply:
quick ques: fault_tolerant doesn't restart/reset the states for `.validate/.test/.predict` call?

Reply:
It does, see the super implementation.

The bug described here is only for when it is called more than once